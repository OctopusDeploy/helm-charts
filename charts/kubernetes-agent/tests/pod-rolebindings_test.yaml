suite: "pod permissions"
templates:
- templates/pod-rolebindings.yaml
tests:
- it: "creates a role binding per target namespace"
  set:
    scriptPods:
      serviceAccount:
        targetNamespaces: ["ns-1", "ns-2", "ns-3"]
    agent.deploymentTarget.enabled: true
  asserts:
  - hasDocuments:
      count: 4

- it: "matches snapshot with one target namespace"
  set:
    scriptPods:
      serviceAccount:
        targetNamespaces: ["ns-1"]
    agent.deploymentTarget.enabled: true
  asserts:
  - matchSnapshot: {}

- it: "only creates the default rolebinding when target namespaces is empty"
  set:
    scriptPods:
      serviceAccount:
        targetNamespaces: []
    agent.deploymentTarget.enabled: true
  asserts:
  - hasDocuments:
      count: 1
  - equal: 
        path: metadata.namespace
        value: NAMESPACE

- it: "only creates the default rolebinding by default"
  set:
    agent.deploymentTarget.enabled: true
  asserts:
  - hasDocuments:
      count: 1
  - equal:
      path: metadata.namespace
      value: NAMESPACE

- it: "is not created when watchdog is disabled"
  set:
    persistence:
      nfs:
        watchdog:
          enabled: false
    agent.deploymentTarget.enabled: true
  asserts:
    - hasDocuments:
        count: 0
        
- it: "creates 1 role-bindings when DeploymentTarget is disabled"
  set:
    agent.deploymentTarget.enabled: false
    scriptPods.serviceAccount.targetNamespaces: ["ns-1", "ns-2", "ns-3"]
    persistence.nfs.watchdog.enabled: true
  asserts:
    - hasDocuments:
        count: 1