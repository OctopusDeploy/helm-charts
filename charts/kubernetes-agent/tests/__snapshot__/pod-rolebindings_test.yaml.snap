matches snapshot with one target namespace:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: octopus-agent-pod-RELEASE-NAME-binding
      namespace: ns-1
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: octopus-agent-pod-RELEASE-NAME-role
    subjects:
      - kind: ServiceAccount
        name: octopus-agent-pod
        namespace: NAMESPACE
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: octopus-agent-pod-RELEASE-NAME-deleter-binding
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: octopus-agent-pod-RELEASE-NAME-delete-role
    subjects:
      - kind: ServiceAccount
        name: octopus-agent-pod
        namespace: NAMESPACE
